docs/kbs_attestation_protocol.md

```diff
@@ -1,220 +1,418 @@
+     ┃                └───────────────────┘              ┃
+     ┃                     ┌────────┐                    ┃
+     ┃◀────────────────────│Response│────────────────────┃
+     ┃      ┌──────────────┴────────┴───────────────┐    ┃
+     ┃      │Header: "Set-Cookie kbs-session-id=123"│    ┃
+     ┃      │Body: Challenge                        │    ┃
+     ┃      └───────────────────────────────────────┘    ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                ┌───────────────────┐              ┃
+     ┃────────────────│POST /kbs/v0/attest│─────────────▶┃
+     ┃        ┌───────┴───────────────────┴───────┐      ┃
+     ┃        │Header: "Cookie kbs-session-id=123"│      ┃
+     ┃        │Body: Attestation                  │      ┃
+     ┃        └───────────────────────────────────┘      ┃
+     ┃                     ┌────────┐                    ┃
+     ┃◀────────────────────│Response│────────────────────┃
+     ┃      ┌──────────────┴────────┴───────────────┐    ┃
+     ┃      │Header: "Set-Cookie kbs-session-id=123"│    ┃
+     ┃      │Body: N/A                              │    ┃
+     ┃      └───────────────────────────────────────┘    ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ┃                                                   ┃
+     ▽                                                   ▽
+               Figure 1: KBS Authentication Phase
+
+```
+
+## Resource Requests
+
+The [authentication](#authentication) phase is the first step of the overall KBS
+protocol, where attesters authenticate themselves against a KBS implementation.
+This leads to the second phase, during which authenticated attesters can request
+protected resources from the KBS.
+
+The KBS implementation keeps track of attestation results and binds them to a
+cookie identifier. During the second phase, the KBS can then decide if a
+specific resource could be released to a given attester, by mapping the cookie
+identifier the attester includes in its resource request message to its
+attestation results.
+
+To request a protected resource from the KBS, the attester sends a `GET` request
+to a resource specific endpoint. If the attester is allowed to access the
+resource, the KBS will respond to the `GET` request with an HTTP response which
+content is set to a [KBS Response](#response) JSON payload:
+
+```
+┌──────────┐                                         ┌─────┐
+│ Attester │                                         │ KBS │
+└───┳──────┘                                         └──┳──┘
+    ┃            ┌─────────────────────────┐            ┃
+    ┃────────────│GET /kbs/v0/<resource_id>│───────────▶┃
```

Comment:
What if `<resource_id>`  is `auth` or `attest`? Could that be confusing to clients or KBS programmers, and hence potentially dangerous from a security pov? I see that this endpoint requires `GET` and those "privileged" commands are using `POST`, but would it be safer if this endpoint was different to the "privileged" one (maybe `/kbs/v0/request/<resource_id>` or siimilar?)

Reply:
I added a `resources/` prefix. And it can either be `resources/key/keyid` or `resources/token/`